hack/multikueue-e2e-test.sh

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SOURCE_DIR="$(cd "$(dirname -- "${BASH_SOURCE[0]}")" && pwd -P)"
+ROOT_DIR=$SOURCE_DIR/..
+export KUSTOMIZE=$ROOT_DIR/bin/kustomize
+export GINKGO=$ROOT_DIR/bin/ginkgo
+export KIND=$ROOT_DIR/bin/kind
+export YQ=$ROOT_DIR/bin/yq
+export E2E_TEST_IMAGE=gcr.io/k8s-staging-perf-tests/sleep:v0.0.3
+export LEADER_KIND_CLUSTER_NAME=${KIND_CLUSTER_NAME}-leader
+export WORKER_KIND_CLUSTER_NAME=${KIND_CLUSTER_NAME}-worker
```

Comment:
I would suggest adding 2 worker clusters in this setup. No reason to go beyond 2 workers, but I think we need 2 to test some important scenarios, for example, 
- to make sure a job does not start at 2 workers at the same time
- the prebuilt workloads are removed from the other workers.

Reply:
I think it is a fair concern, I just tested locally creating 3 clusters and it worked, but I see this might be an issue. If you had to increase the limits to run the tests, I think we could add a comment referring to the potential need to increase the limits. Not clear how often does this happen to me at this moment.

Reply:
There is a comment in the most likely place to fail (when waiting for kueue to start).